pandas/core/tools/datetimes.py

```diff
@@ -434,9 +449,11 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                     yearfirst=yearfirst,
                     require_iso8601=require_iso8601
                 )
-
             if is_datetime64_dtype(result) and box:
                 result = DatetimeIndex(result, tz=tz, name=name)
+            # GH 6415
+            elif arg_is_series and utc:
```

Comment:
I *think* (didn't look in detail) that this kludgyness is coming from the fact that for a Series `_convert_listlike` now converts an array (`box` is to False when dealing with a Series), while for DatetimeIndex is returns a DatetimeIndex. And, an array cannot hold tz aware data, so the `utc` keyword needs to be handles separately for Series and DatetimeIndex, when using this current implementation.

But, can't we just remove the `box=False` for converting Series objects? And so unify the code for returning Series vs DatetimeIndex (conversion from DatetimeIndex to Series (instead of from array to Series) should be rather performant?)